crates/katana/executor/src/abstraction.rs

```diff
@@ -192,6 +194,9 @@ pub trait TransactionExecutionOutput {
 
     /// The error message if the transaction execution reverted, otherwise the value is `None`.
     fn revert_error(&self) -> Option<&str>;
+
+    /// Retrieves the execution info of the transaction.
+    fn execution_info(&self) -> TxExecInfo;
```

Comment:
lets return a & instead, we should do the same for receipt as well.. and let the user to choose whether to clone it or not.

Reply:
Same for the receipt, we should implement a cache system to return a reference to the cache? Because as the receipt is fully constructed when the function is called, can't easily return a reference at the moment?

Should we consider this here, or in a PR dedicated for those improvements?